### ISSUE TITLE
MAE is available in HuggingFace Transformers (both PyTorch and TF)

### ISSUE THREAD

**Issue by NielsRogge** [facebookresearch/mae#103]
Hi everyone!

Thanks for this great work. Since several months, MAE is available in HuggingFace Transformers with an easy-to-use and familiar API: https://huggingface.co/docs/transformers/model_doc/vit_mae

The model is available both in PyTorch and TensorFlow.

The API is the same as `BertModel` for instance (but it's called `ViTMAEModel`). There's also a `ViTMAEForPreTraining` class, similar to `BertForPreTraining`, which includes the decoder and loss calculation.

We do provide an easy-to-use script to directly pre-train a ViTMAE model on your custom data, called `run_mae.py`. It can be found here: https://github.com/huggingface/transformers/tree/main/examples/pytorch/image-pretraining. The script leverages the 🤗 Trainer and runs directly on a distributed environment. The script also allows for logging to Tensorboard/Weights and Biases, and pushing the model to the hub during and/or after training (with easy reloading afterwards using `from_pretrained`).

After pre-training, you can automatically load the weights into a `ViTForImageClassification` model, allowing for fine-tuning. Fine-tuning notebooks can be found [here](https://github.com/huggingface/notebooks/blob/main/examples/image_classification.ipynb) (for Torchvision for data augmentation) or [here](https://github.com/huggingface/notebooks/blob/main/examples/image_classification_albumentations.ipynb) (in case you prefer Albumentations for data augmentation).

I also created a notebook to illustrate visualization (based on this repo): https://github.com/NielsRogge/Transformers-Tutorials/blob/master/ViTMAE/ViT_MAE_visualization_demo.ipynb

**Comment by ross-Hr** [comment]
Great!!!

**Comment by AnukritiSinghh** [comment]
hi, thanks for this!!
can I use this to fine-tune on my custom dataset but instead of classification I want to do reconstruction only ?

**Comment by NielsRogge** [comment]
Yes, you could do that, although the reconstruction objective is purely used to learn good image features. The reconstructions themselves are not really useful, and also blurry as can be seen in the notebook above.

**Comment by surajraj99** [comment]
@NielsRogge Do the model weights on the hub use norm_pix_loss = False? It seems like it does based on the config files: https://huggingface.co/facebook/vit-mae-large/blob/main/config.json.

I don't have the resources to retrain the Vit from scratch using norm_pix_loss = True for tensorflow so I was hoping to use hugginface's models.

I am wondering if there is a pretrained on imagenet model weights available where norm_pix_loss is True? I am trying to do downstream tasks in tensorflow and it seems like having norm_pix_loss True is better for that. Is that available?

**Comment by geonuk-kimmm** [comment]
Thanks for the sharing!


But following ipynb files are not available.
https://github.com/huggingface/notebooks/blob/main/examples/image_classification.ipynb
https://github.com/huggingface/notebooks/blob/main/examples/image_classification_albumentations.ipynb
https://github.com/NielsRogge/Transformers-Tutorials/blob/master/ViTMAE/ViT_MAE_visualization_demo.ipynb

I hope it could be available soon.

**Comment by NielsRogge** [comment]
See my comment here: https://github.com/NielsRogge/Transformers-Tutorials/issues/489#issuecomment-2804218448
